src/EprPrnIntegration.Api/FetchNpwdIssuedPrnsFunction.cs

```diff
@@ -56,23 +85,117 @@ public async Task Run([TimerTrigger("%FetchNpwdIssuedPrns:Schedule%")] TimerInfo
             }
             catch (Exception ex)
             {
-                _logger.LogError(ex,$"Failed Get Prns method for filter {filter} with exception {ex}", filter, ex.Message);
+                _logger.LogError(ex, $"Failed Get Prns method for filter {filter} with exception {ex}", filter, ex.Message);
                 throw;
             }
 
-
             try
             {
                 await _serviceBusProvider.SendFetchedNpwdPrnsToQueue(npwdIssuedPrns);
+
                 _logger.LogInformation("Issued Prns Pushed into Message Queue");
+
+                await _utilities.SetDeltaSyncExecution(deltaRun, toDate);
             }
             catch (Exception ex)
             {
                 _logger.LogError("Failed pushing issued prns in message queue with exception: {ex}", ex);
                 throw;
             }
 
+            try
+            {
+                await ProcessIssuedPrnsAsync();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Failed fetching prns from the queue with exception: {ex}", ex);
+                throw;
+            }
+
             _logger.LogInformation($"FetchNpwdIssuedPrnsFunction function Completed at: {DateTime.UtcNow}");
         }
+
+        public async Task ProcessIssuedPrnsAsync()
```

Comment:
this should be private method since we are not using it anywhere else. if its for testing, we can have two options, either move this to its own helper class and wrap it with interface so it can be tested and mocked in isolation, or make it internal and in the test project give the permission to access to internal methods/classes

Reply:
Done